### DISCUSSION TITLE
Limit parameter store requests to the default context and profile.

### DISCUSSION THREAD

**Discussion by niels1voo** [awspring/spring-cloud-aws#433] [Q&A] [answered]
using version 2.4.1, wondering if it is possible to configure the property source to avoid making (3/4) calls to paths that don't interest us?

We have the following parameter store configuration:
aws:
  paramstore:
    default-context: our-context
    enabled: true
    prefix: /our-prefix
    profile-separator: '/'

spring_profiles_active=env
In our application's configuration we are only interested in the configurations set at
**/our-prefix/our-context/env**, the other paths are empty.

Unfortunately, this is the last path that is requested, following three useless calls, contributing to  parameter store throttling exceptions when we spin up multiple fargate containers.

_**useless preliminary calls**_
_19:19:11.873 [main] INFO  i.a.c.p.AwsParamStorePropertySourceLocator - Loading property from AWS Parameter Store with name: /our-prefix/application/, optional: false
19:19:13.993 [main] INFO  i.a.c.p.AwsParamStorePropertySourceLocator - Loading property from AWS Parameter Store with name: /our-prefix/application/env/, optional: false
19:19:14.085 [main] INFO  i.a.c.p.AwsParamStorePropertySourceLocator - Loading property from AWS Parameter Store with name: /our-prefix/our-context/, optional: false_
**the one we care about**
**19:19:15.899 [main] INFO  i.a.c.p.AwsParamStorePropertySourceLocator - Loading property from AWS Parameter Store with name: /our-prefix/our-context/env/, optional: false**

**Comment by maciejwalkowiak** [comment, marked as answer]
Have you considered using `spring.config.import=aws-parameterstore:/our-prefix/our-context/env` instead of the legacy Spring Cloud Bootstrap context? This will load only what you exactly specify in configuration.

**Reply by niels1voo** [reply]
Thanks for the tip! This worked. 
One minor correction, I needed a slash following the env. 

I had considered this technique, but the [documentation](https://docs.awspring.io/spring-cloud-aws/docs/2.4.1/reference/html/index.html#spring-boot-configuration-import-support) doesn't show that you can do this, it only mentions appending specific parameters after the colon.

**Reply by maciejwalkowiak** [reply]
Indeed this can be improved. Would you like to make a PR?

**Reply by niels1voo** [reply]
Just saw your suggestion. If this hasn't been done yet I will take it on.
